test/any/CanonicalizerRunner-test.js

```diff
@@ -0,0 +1,139 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+'use strict';
+
+describe('CanonicalizerRunner', function() {
+  var CanonicalizerRunner;
+  var IDLFile;
+  var IDLFileContents;
+  var Parser;
+  var PipelineMessage;
+  var ResultBox;
+
+  beforeEach(function() {
+    foam.CLASS({
+      package: 'org.chromium.webidl.Test',
+      name: 'ResultBox',
+      implements: ['foam.box.Box'],
+
+      properties: [
+        {
+          class: 'Array',
+          name: 'results',
+        },
+      ],
+
+      methods: [
+        function send(msg) {
+          this.results.push(msg);
+        },
+        function clear() {
+          this.results = [];
+        }
+      ]
+    });
+
+    CanonicalizerRunner = foam.lookup('org.chromium.webidl.CanonicalizerRunner');
+    IDLFile = foam.lookup('org.chromium.webidl.IDLFile');
+    IDLFileContents = foam.lookup('org.chromium.webidl.IDLFileContents');
+    Parser = foam.lookup('org.chromium.webidl.Parser');
+    PipelineMessage = foam.lookup('org.chromium.webidl.PipelineMessage');
+    ResultBox = foam.lookup('org.chromium.webidl.Test.ResultBox');
+  });
+
+  it('should send an error if invalid arguments are received as message', function() {
```

Comment:
This test pattern crops up with every `PipelineRunner` implementation. Should we write a `-helper.js` function that can be used to capture "wrong obj" and "missing input data" cases?